Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs

```diff
@@ -300,38 +301,69 @@ public Task PutStake(IEnumerable<StakeItem> stakeEntries)
         /// <summary>
         /// Persists unsaved POS blocks information to the database.
         /// </summary>
-        /// <param name="stakeEntries">List of POS block information to be examined and persists if unsaved.</param>
-        private void PutStakeInternal(IEnumerable<StakeItem> stakeEntries)
+        /// <param name="stakeItems">List of POS block information to be examined and persisted, if unsaved.</param>
+        private void PutStakeInternal(IEnumerable<StakeItem> stakeItems)
         {
-            foreach (StakeItem stakeEntry in stakeEntries)
+            foreach (StakeItem stakeItem in stakeItems)
             {
-                if (!stakeEntry.InStore)
+                if (!stakeItem.ExistsInStore)
                 {
-                    this.session.Transaction.Insert<byte[], BlockStake>("Stake", stakeEntry.BlockId.ToBytes(false), stakeEntry.BlockStake);
-                    stakeEntry.InStore = true;
+                    this.session.Transaction.Insert<byte[], BlockStake>("Stake", stakeItem.BlockHash.ToBytes(false), stakeItem.BlockStake);
+                    stakeItem.InStore();
                 }
             }
         }
 
         /// <summary>
-        /// Retrieves POS blocks information from the database.
+        /// Retrieves a POS block from the database.
+        /// </summary>
+        /// <param name="stakeItem">Partially initialized <see cref="StakeItem"/> that is populated by <see cref="BlockStake"/>.</param>
+        public Task GetStakeItem(StakeItem stakeItem)
+        {
+            return this.session.Execute(() =>
+            {
+                this.logger.LogTrace("({0}:{1})", nameof(stakeItem.BlockHash), stakeItem.BlockHash);
+
+                var blockStake = this.session.Transaction.Select<byte[], BlockStake>("Stake", stakeItem.BlockHash.ToBytes(false));
+                if (!blockStake.Exists)
+                    this.logger.LogTrace("{0}:{1} [NOT IN DATABASE]", nameof(stakeItem.BlockHash), stakeItem.BlockHash);
```

Comment:
I know I've still did not create the how to write logging document, so it is not really your fault, but this is not what I'm looking for in logging. So I'll try to explain.

There are two special kinds of logs - entry and exit logs in methods. Almost every method should have them except for methods that are called too often. And by too often I rather mean relative count. I.e. we can have a block with 1000 transactions and it is perfectly OK if a complex method that works with transaction logs on entry and exit for each transaction. Too often is when for a single object we have a for loop which executes trivial methods.

Entry logs come in pattern `"(arg1:value1,arg2:value2,...)"`, exit logs come in pattern `"(-):resultValue,outarg1=outval1,outarg2=outval2,..."`. And we are looking for every entry log to have its matching exit log. Recently, I've had a crash and there was not full stack info, so I did not know where exactly the crash happened. But I did have exit logs for every entry logs except for the final path to the crashing line -> easy find.

Now if a method has a multiple ways of exit (multiple returns, exceptions) then we want to distinguish in log, which return it was, in that case the exit log pattern extends to: `"(-)[UNIQUE_CONTEXT]:resultValue,outarg1=outval1,outarg2=outval2,..."`

Now with both entry and exit logs we use nameof() to put names of arguments in the log.
And finally, entry and exit logs are made on trace level.

OK, so those are entry and exit logs, but we also log interesting information inside the methods. We want those to look differently (visually) compared to entry/exit. 
Their pattern is `"Not very complex, but unique, English sentence."`. We do not use '[CONTEXT]' as the context is given by the sentence itself. We also do not use nameof() because we rather describe 
the event in English. We also don't use `"name:value"` notation or `"name=value"`.

These logs can be on trace level, debug level or info level, or even warning or error/fatal level.

So, instead of 

```
this.logger.LogTrace("{0}:{1} [NOT IN DATABASE]", nameof(stakeItem.BlockHash), stakeItem.BlockHash); 
```

We are looking for something like this:
``` 
this.logger.LogTrace("Item hash '{0}' not found in the database.", stakeItem.BlockHash); 
```

Whether or not to enclose a value in `'{0}'` depends on its type. All small integer types, booleans etc. go without apostrophes, strings, block hashes or hash values, or more complex values do, except for arrays that are `'[val1,val2,...]'`.

Similarly
```
this.logger.LogTrace("{0} [NULL]", nameof(stakeItem.BlockStake));
```

->

```
this.logger.LogTrace("Stake information is mossing for item hash '{0}'.", stakeItem.BlockHash);
```